src/Parser/AfflictionWarlock/Modules/SoulShards/SoulShardBreakdown.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+class SoulShardBreakdown extends React.Component {
+  static propTypes = {
+    shardsGained: PropTypes.object.isRequired,
+    shardsSpent: PropTypes.object.isRequired,
+    shardsWasted: PropTypes.object.isRequired,
+  };
+  prepareGenerated(shardGen, shardWasted){
+    //shardGen and shardWasted has the same number of abilities (some having 0, these can be now filtered)
+    const abilities = [];
+    Object.keys(shardGen).map(abilityId => abilities.push({abilityId: Number(abilityId), generated: shardGen[abilityId].shards, wasted: shardWasted[abilityId].shards}));
+    abilities.sort((a, b) => b.generated - a.generated);
+    return abilities;
```

Comment:
Come to think of it, instead of map, forEach would be correct (but I'm not very experienced in Javascript and this worked, that's why it's in there). But if you say this works too, I shall use your version.